lib/jekyll-import/importers/rss.rb

```diff
@@ -89,8 +93,28 @@ def self.write_rss_item(item, options)
           end
 
           f.puts output
+          if canonical_link
+            f.puts <<~HTML
+              <p>Originally posted on <a href="#{item.link}">#{URI.parse(item.link).host}</a>.</p>
```

Comment:
I still think we should remove this before merging. What if the blog is not in English? Or if they just don't want this link? Now the user has to manually edit all of their posts. This is asking way more of the user than editing their post layout. In general, injecting our own *content* is not what we do. Metadata is fine but I'm not comfortable writing content to these posts.

> it's configurable using `--canonical_link` option

True but then you lose the front matter value. The docs for this only says you get the front matter value, not extra content. We can trust that the user can add a link themselves if they so desire. Adding canonical_url is sufficient.

Reply:
Hi @parkr, Your point is valid in terms of user can add it if they want. But I am not sure how it can be automated?

For my case, I import all my posts from Medium to github pages. I want to add **canonical_url** at the end of the imported content. You can see the [example here](https://sumanmaity112.github.io/posts/is-it-the-era-of-mono-repo/#conclusion). Currently, I have a [Github Workflow](https://github.com/sumanmaity112/sumanmaity112.github.io/blob/main/.github/workflows/import.yml) to import all posts from Medium. 
At least I am not sure how to achieve this if the importer is not adding it automatically. Maybe you can help me here. Thanks

Reply:
Cool workflow! Nice way of having a backup/mirror of your site.

I mean that adding it to the post layout adds this link to every post automatically since every post used the post layout, right? Even if they don't, you could add front matter defaults so that each post uses the post layout, then add the link to the post layout.

1. Post itself contains only the content the user wrote
2. _layouts/post.html contains canonical link if specified
3. Each post specifies that it should use the post layout, or use front matter defaults to specify the post layout should be used by all posts

I don't want this importer to modifying the user's post contents (especially when it's not documented).

Reply:
@parkr Your point is valid.

Now in my mind there are couple of options
1. Remove the changes which adds canonical url to the post
2. Keep the changes and document it
3. Add a new options (addition to `--canonical_link`) to configure this specific footer content

Please let me know whatever you think is better.

Reply:
@sumanmaity112 Since Parker doesn't want importers to modify *imported content* regardless of configuration, it's best that you remove the `canonical_url` **markup injection**. Let the option to add the info to front matter remain for subsequent consumption.

Regarding your personal blog, you have two options:
1. Modify the footer template in your theme to render the canonical url.
2. Write a custom Ruby plugin into a `_plugins` directory in your blog source repository. The plugin can easily inject custom content into your posts at runtime via Jekyll Hooks.